src/compiler/checker.ts

```diff
@@ -4048,7 +4048,10 @@ namespace ts {
                         context.flags ^= NodeBuilderFlags.InInitialEntityName;
                     }
                     let firstChar = symbolName.charCodeAt(0);
-                    const canUsePropertyAccess = isIdentifierStart(firstChar, languageVersion);
+                    const canUsePropertyAccess = firstChar === CharacterCodes.hash ?

```

Comment:
Probably best to check the name node itself rather than rely on checking for `#`. Something like:
isNamedDeclaration(symbol.valueDeclaration) ? isPrivateName(symbol.valueDeclaration.name)

Reply:
I think I have to use the symbol name because apparently `valueDeclaration` can be undefined. Making the change to check `isNamedDeclaration(valueDeclaration)` crashes tons of tests because it's undefined.

Reply:
{vw}